apps/heureka/src/components/tabs/TabContext.jsx

```diff
@@ -37,15 +37,22 @@ const TabContext = () => {
   const { setActiveTab } = useGlobalsActions()
   const activeTab = useGlobalsActiveTab()
 
-  const memoizedTabs = useMemo(
+  // Memorized top navigation items
+  const memorizedTabs = useMemo(
     () =>
       TAB_CONFIG.map((tab) => (
-        <TabNavigationItem key={tab.value} icon={tab.icon} label={tab.label} value={tab.value} />
+        <TopNavigationItem
+          key={tab.value}
+          label={tab.label}
+          active={activeTab === tab.value} // Set the active item
+          onClick={() => setActiveTab(tab.value)} // Trigger tab change
+        />
       )),
-    []
+    [activeTab, setActiveTab]
   )
 
-  const memoizedTabPanels = useMemo(
+  // Memorized tab panels
+  const memorizedTabPanels = useMemo(
     () =>
       TAB_CONFIG.map((tab) => (
         <TabPanel key={tab.value} value={tab.value}>
```

Comment:
You shouldn't be using the tab panel anymore after the switch to TopNavigation

Reply:
why? as the tabPanel contains the div for each top nav item content (its children) should be there. I can merge it into NavEntryContext but I rather prefer to keep it in separated control as it is now.